```bash
=========================== short test summary info ============================
FAILED pandas/tests/io/test_parquet.py::TestBasic::test_parquet_read_from_url[pyarrow] - AssertionError: DataFrame.columns are different

Attribute "dtype" are different
[left]:  object
[right]: StringDtype(storage=pyarrow, na_value=nan)
```

--- a/pandas/tests/io/test_parquet.py
+++ b/pandas/tests/io/test_parquet.py
@@ -683,7 +683,11 @@ class TestBasic(Base):
         with open(datapath("io", "data", "parquet", "simple.parquet"), mode="rb") as f:
             httpserver.serve_content(content=f.read())
             df = read_parquet(httpserver.url, engine=engine)
-        tm.assert_frame_equal(df, df_compat)
+
+        expected = df_compat
+        if pa_version_under19p0:
+            expected.columns = expected.columns.astype(object)
+        tm.assert_frame_equal(df, expected)
 
 
 class TestParquetPyArrow(Base):

